pkg/asset/installconfig/alibabacloud/validation.go

```diff
@@ -16,76 +15,108 @@ func Validate(client *Client, ic *types.InstallConfig) error {
 	platformPath := field.NewPath("platform").Child("alibabacloud")
 	allErrs = append(allErrs, validatePlatform(client, ic, platformPath)...)
 
-	if ic.ControlPlane != nil && ic.ControlPlane.Platform.AlibabaCloud != nil {
-		allErrs = append(allErrs, validateMachinePool(client, ic, field.NewPath("controlPlane", "platform", "alibabacloud"), ic.ControlPlane.Platform.AlibabaCloud, ic.ControlPlane.Replicas)...)
+	allErrs = append(allErrs, validateControlPlaneMachinePool(client, ic)...)
+	allErrs = append(allErrs, validateComputeMachinePool(client, ic)...)
+
+	return allErrs.ToAggregate()
+}
+
+func validateControlPlaneMachinePool(client *Client, ic *types.InstallConfig) field.ErrorList {
+	allErrs := field.ErrorList{}
+	var mpool mergedMachinePool
+	defaultPool := alibabacloudtypes.DefaultMasterMachinePoolPlatform()
+	mpool.SetWithFieldPath(&defaultPool, field.NewPath("default"))
```

Comment:
@staebler I added a `mergedMachinePool` type to store the field path, but there is still one unsatisfactory place, which is the use of `field.NewPath("default")`

Reply:
No, the "default" field path is not acceptable.

Ideally, the default value should never be the cause of the validation failure. If we use the default value and it fails validation, then there should be some other field that caused that validation instead. For example, if the user did not specify the instance type, but the default instance type is not available in the region that the user specified, then the validation error should ideally be on the region rather than the instance type.

However, I understand that the gets pretty tricky, so I can accept the causes of the validation failures as we have them currently. If we use the default value and it fails validation, then let's attribute that to the specific machine pool. For example, if the default instance type fails validation for the control plane, then the field path reported in the error should be "controlPlane.platform.alibabacloud.instanceType".

Reply:
> For example, if the default instance type fails validation for the control plane, then the field path reported in the error should be "controlPlane.platform.alibabacloud.instanceType".

Okay, thank you, I used the machine pool field path for the default value.

Reply:
@staebler Can you review again？